loader/dd_library_loader.c

```diff
@@ -69,13 +139,36 @@ static void ddtrace_pre_minit_hook(void) {
         }
 
         // Set 'datadog.trace.sources_path' setting
-        zend_string *name = ddloader_zend_string_init(php_api_no, ZEND_STRL("datadog.trace.sources_path"), 1);
-        zend_string *value = ddloader_zend_string_init(php_api_no, sources_path, strlen(sources_path), 1);
+        ddloader_ini_set_configuration(ZEND_STRL("datadog.trace.sources_path"), sources_path, strlen(sources_path));
         free(sources_path);
+    }
+
+    // Load, but disable the tracer if runtime configuration is not safe for auto-injection
+    bool disable_tracer = false;
+
+    char *incompatible_exts[] = {"Xdebug", "the ionCube PHP Loader", "newrelic", "blackfire", "pcov"};
+    for (size_t i = 0; i < sizeof(incompatible_exts) / sizeof(incompatible_exts[0]); ++i) {
+        if (ddloader_is_ext_loaded(incompatible_exts[i])) {
+            if (force_load) {
+                LOG(WARN, "Potentially incompatible extension detected: %s. Ignoring as DD_INJECT_FORCE is enabled", incompatible_exts[i]);
+            } else {
+                LOG(WARN, "Potentially incompatible extension detected: %s. ddtrace will be disabled unless the environment DD_INJECT_FORCE is set to '1', 'true', 'yes' or 'on'", incompatible_exts[i]);
+                disable_tracer = true;
+            }
+        }
+    }
 
-        zval tmp;
-        ZVAL_STR(&tmp, value);
-        ddloader_zend_hash_update(configuration_hash, name, &tmp);
+    if (ddloader_is_opcache_jit_enabled()) {
+        if (force_load) {
+            LOG(WARN, "OPcache JIT is enabled and may cause instability. Ignoring as DD_INJECT_FORCE is enabled");
+        } else {
+            LOG(WARN, "OPcache JIT is enabled and may cause instability. ddtrace will be disabled unless the environment DD_INJECT_FORCE is set to '1', 'true', 'yes' or 'on'");
+            disable_tracer = true;
+        }
+    }
+
+    if (disable_tracer) {
+        ddloader_ini_set_configuration(ZEND_STRL("datadog.trace.enabled"), ZEND_STRL("0"));
```

Comment:
You actually want `ddtrace.disable=1`. `datadog.trace.enabled=0` still prepares most of the process (setting hooks, etc., just skips the actual tracing part).